Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/MSIBannerNotificationServiceTest.cs

```diff
@@ -0,0 +1,92 @@
+
+            _fakeBannerNotificationConfiguration.Value.IsBannerNotificationEnabled = true;
+
+            _mSIBannerNotificationService = new MSIBannerNotificationService(_fakeCacheConfiguration, _fakeAzureStorageService, _fakeAzureTableStorageClient, _fakeBannerNotificationConfiguration, _fakeLogger);
+        }
+
+        [Test]
+        public async Task WhenIsBannerNotificationDisabled_ThenNullValueForBannerNotificationMessageIsAssigned()
+        {
+            _fakeBannerNotificationConfiguration.Value.IsBannerNotificationEnabled = false;
+
+            string result = await _mSIBannerNotificationService.GetBannerNotification(string.Empty);
+
+            Assert.IsNull(result);
+        }
+
+        [Test]
+        public async Task WhenExceptionThrownByService_ThenNullValueForBannerNotificationMessageIsAssigned()
+        {
+            A.CallTo(() => _fakeAzureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Throws(new Exception());
+
+            string result = await _mSIBannerNotificationService.GetBannerNotification(string.Empty);
+
+            Assert.IsNull(result);
+        }
+
+        [Test]
+        public async Task WhenBannerNotificationNotFound_ThenEmptyValueForBannerNotificationMessageIsAssigned()
+        {
+            A.CallTo(() => _fakeAzureTableStorageClient.GetSingleEntityAsync(A<string>.Ignored, A<string>.Ignored)).MustNotHaveHappened();
+
+            string result = await _mSIBannerNotificationService.GetBannerNotification(string.Empty);
+
+            Assert.IsEmpty(result);
+        }
+
+        [Test]
+        public async Task WhenBannerNotificationFoundWithXSS_ThenValueExcludingXSSForBannerNotificationMessageIsAssigned()
+        {
+            string message = @"This system will be under maintenance. For more details <a href=""https://www.test.com"" target=""_blank"">click here<script>alert('XSS attack!')</script>";
+            string expectedMessage = @"This system will be under maintenance. For more details <a href=""https://www.test.com"" target=""_blank"">click here</a>";
+
+            A.CallTo(() => _fakeAzureTableStorageClient.GetSingleEntityAsync(A<string>.Ignored, A<string>.Ignored)).Returns(new MsiBannerNotificationEntity() { Message = message });
+
+            string result = await _mSIBannerNotificationService.GetBannerNotification(string.Empty);
+
+            Assert.AreEqual(expectedMessage, result);
+        }
+
+        [Test]
+        public async Task WhenBannerNotificationFound_ThenValueForBannerNotificationMessageIsAssigned()
+        {
+            A.CallTo(() => _fakeAzureTableStorageClient.GetSingleEntityAsync(A<string>.Ignored, A<string>.Ignored)).Returns( new MsiBannerNotificationEntity() { Message ="test"});
+
+            string result =  await _mSIBannerNotificationService.GetBannerNotification(string.Empty);
+
+            Assert.AreEqual("test", result);
+        }
+    }
```

Comment:
Should there be another test for selecting and displaying the first entry if more than one is found in the table?

This would reflect the ACs.

It is worth noting that I have seen the code that would deal with this, but it should be tested.

Reply:
To write this test case should not be possible because 
**_azureTableStorageClient.GetSingleEntityAsync()** method internally handles return only one latest start date banner notification message.
That method we already excluded from code coverage.

Reply:
Not ideal, but I can see why